### DISCUSSION TITLE
How to upgrade postgres in docker compose?

### DISCUSSION THREAD

**Discussion by Snawe** [muety/wakapi#725] [Q&A]
Hi,

I am using docker-compose to run wakapi.
With this change https://github.com/muety/wakapi/commit/2568fbd39bd48b6a8d566c7b9ed8dcefafa28430 the postgres version was bumped to 17.
Can anyone tell me a way to migrate the version 12 data to 17? I tried couple of articles I found online, but they are either not for containers or seem to be unsafe or use `pg_upgrade` which I am just to dump to use in the docker container :D

**Comment by muety** [comment]
I successfully migrated my local dev instance's database from 12 to 17 using `pg_upgrade`, roughly following [this guide](https://medium.com/@mpyatishev/seamless-postgresql-upgrade-migrating-from-version-12-to-16-in-docker-8141a8905899).

**Reply by Snawe** [reply]
Hmmm, could you maybe pleas elaborate on "roughly"?
I tried to follow the guide and got some 

```
connection to server on socket "/var/lib/postgresql/.s.PGSQL.50432" failed: FATAL:  role "postgres" does not exist
```

I than created the postgres user on both the volume using postgres 17 and postgres 12.
It now basically fails with

```
$ docker run --rm -v wakapi_wakapi-db-data:/var/lib/postgresql/12/data -v wakapi_wakapi-db-data-v17:/var/lib/postgresql/17/data tianon/postgres-upgrade:12-to-17
Performing Consistency Checks
-----------------------------
Checking cluster versions                                     ok
Checking database user is the install user                    
database user "postgres" is not the install user
Failure, exiting
```

**Reply by muety** [reply]
Can't remember the exact process, but there should be plenty of guides on the web. If you used the default compose file, the database user is `wakapi`.

**Reply by Snawe** [reply]
yeah, I know that the default user is `wakapi`. 
Anyhow. Could not make it work. Will stick to postgres 12 than. Did spend another 2 hours trying to move permissions around and make the upgrade work with `wakapi`. Couldn't :shrug: 

It's running fine with postgres 12 for now. So fine for me :D Ty for your help and all the amazing work :)

If someone else finds out how to do it, I am open for a link ^^
